catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/FederationAdminServiceImpl.java

```diff
@@ -431,6 +452,206 @@ public void init() {
+            String message = "Error querying for subscribed metacards;";
+            LOGGER.error("{} For source ids: {}", message, pollableSourceIds);
+            throw new FederationAdminException(message, e);
+        }
+
+        return remoteRegistryMetacards;
+    }
+
+    private Map<String, Metacard> getNonLocalRegistryMetacardsMap()
+            throws FederationAdminException {
+
+        Map<String, Metacard> nonLocalMetacardsMap = new HashMap<>();
+        List<Filter> filters = new ArrayList<>();
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.CONTENT_TYPE),
+                RegistryConstants.REGISTRY_NODE_METACARD_TYPE_NAME));
+        filters.add(FILTER_FACTORY.like(FILTER_FACTORY.property(Metacard.TAGS),
+                RegistryConstants.REGISTRY_TAG));
+        filters.add(filterBuilder.not(filterBuilder.attribute(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE)
+                .is()
+                .bool(true)));
+
+        Filter filter = FILTER_FACTORY.and(filters);
+
+        List<Metacard> nonLocalMetacardList = getRegistryMetacardsByFilter(filter);
+
+        for (Metacard metacard : nonLocalMetacardList) {
+            String registryId = metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                    .getValue()
+                    .toString();
+            nonLocalMetacardsMap.put(registryId, metacard);
+        }
+
+        return nonLocalMetacardsMap;
+    }
+
+    private void addOrUpdateMetacardsForRemoteRegistries() throws FederationAdminException {
+        addOrUpdateMetacardsForRemoteRegistries(pollableSourceIds);
+    }
+
+    private void addOrUpdateMetacardsForRemoteRegistries(Set<String> sourceIds)
+            throws FederationAdminException {
+        if (CollectionUtils.isEmpty(sourceIds)) {
+            throw new FederationAdminException(
+                    "Error adding or updating metacard for remote registry. Empty set of source ids provided.");
+        }
+
+        Map<String, Metacard> remoteRegistryMetacardsMap = getRemoteRegistryMetacardsMap();
+        Map<String, Metacard> nonLocalMetacardsMap = getNonLocalRegistryMetacardsMap();
+
+        Map<Serializable, Metacard> remoteMetacardsToUpdate = new HashMap<>();
+        List<Metacard> remoteMetacardsToCreate = new ArrayList<>();
+
+        for (Map.Entry<String, Metacard> remoteEntry : remoteRegistryMetacardsMap.entrySet()) {
+            if (nonLocalMetacardsMap.containsKey(remoteEntry.getKey())) {
+                Metacard existingMetacard = nonLocalMetacardsMap.get(remoteEntry.getKey());
+
+                if (remoteEntry.getValue()
+                        .getModifiedDate()
+                        .after(existingMetacard.getModifiedDate())) {
+                    remoteMetacardsToUpdate.put(remoteEntry.getKey(), remoteEntry.getValue());
+                }
+
+            } else {
+                remoteMetacardsToCreate.add(remoteEntry.getValue());
```

Comment:
Changed to check against all registry metacards. Adding a check for local registry entries